### ISSUE TITLE
`julia .` does not fail

### ISSUE THREAD

**Issue by jakobnissen** [JuliaLang/julia#51791]
If a directory is passed as the script to `julia`, no error is raised, but nothing happens. Possibly a consequence of https://github.com/JuliaLang/julia/issues/36605.
Ideally, this should throw an error.

Example:
```
$ julia .
$
```
More insidious example
```
$ julia --startup=no --project . my_script arg1 arg2 arg3
$
```

**Comment by fredrikekre** [comment]
The program (`.`) is empty
```
julia> read(pwd())
UInt8[]
```
See e.g. #36605.

**Comment by KristofferC** [comment]
I think this can be closed as a dup since this is just a "downstream" effect of #36605

**Comment by jakobnissen** [comment]
I disagree - currently, #36605 is not yet solved because checking whether a file is a directory is an expensive operation. So, that issue may be closed because it's deemed that the check is not worth it.
If it does, then this issue should still be addressed with a `isdir` check, since the overhead of checking whether a file is a directory is insignificant compared to starting Julia.

**Comment by jakobnissen** [comment]
This appears to have been fixed on somewhere between 1.1.0.8 and 1.11.3, perhaps unintentionally 🎉
